### PR TITLE
Fixes/memory leaks

### PR DESCRIPTION
**Pull request by danwalmsley**

### CODE REVIEW

File: src/Mabiavalon.DockNC/Docking/Branch.cs [hunk 1]
```diff
@@ -135,7 +151,7 @@ private void UpdateLogicalChildren(object oldItem, object newItem)
 
             if (oldChild != null)
             {
-                ((ISetLogicalParent) oldItem).SetParent(null);
+                //((ISetLogicalParent)oldItem).SetParent(null);
```
Comment (Seeker1437) [inline src/Mabiavalon.DockNC/Docking/Branch.cs]: Could you explain this change? Wouldn't this cause issues with the logical tree?

**Review by Seeker1437 — COMMENTED** [review]
Does this cause a memory leak? I need to be able to ensure that items that were removed are not still logical children, I see this being an issue because we set the item's parent to the branch first, we would have to remove it from the parent before anything else tries to set the parent. Thus will break `DockControl.Dock()` as well as any attempt to change any of the branch items from code....

File: src/Mabiavalon.DockNC/Docking/Branch.cs (hunk 1 above)
Comment (danwalmsley) [reply src/Mabiavalon.DockNC/Docking/Branch.cs]: Sorry this is WIP

Comment (danwalmsley) [reply src/Mabiavalon.DockNC/Docking/Branch.cs]: This is just a temporary test, waiting to debug this through Avalon studio.

but my theory is you should set the parent to null after removing.. assuming avalonia itself will try to manage this?

**Comment by danwalmsley** [comment]
I'm trying to fix this exception... which is caused by the line you are asking about...
image

File: src/Mabiavalon.DockNC/Docking/Branch.cs (hunk 1 above)
Comment (Seeker1437) [reply src/Mabiavalon.DockNC/Docking/Branch.cs]: Avalonia does not, unfortunately. The strange thing is that this exactly what `ContentControl` does in that order.

**Comment by Seeker1437** [comment]
Can you point me to a repro for that particular exception?

**Comment by danwalmsley** [comment]
OK I corrected code, I don't think it was causing my memory leak, but I double checked in the debugger and avalonia sets and clears the parent for us. so we might have been stopping avalonia handling logical tree correctly.

**Comment by danwalmsley** [comment]
Also this pr adds disposing of subscribed observables. which would potentially leak with out this.

**Pull request merged**